test/README.md

```diff
@@ -0,0 +1,44 @@
+# Linear Vesting Tests Documentation
+
+This offers a comprehensive guide to the Linear Vesting tests within the project. The property based tests (in `./LinearVestingSpec.hs`) are meticulously designed to assess the accuracy of Linear Vesting implementation, particularly focusing on the validation of actions involved during the unlocking phase of vesting.
+
+## Overview
+
+The property based tests for the Linear Vesting are organized into a suite titled "Linear Vesting Tests", This suite is dedicated to verifying the validation process for the accurate implementation of Linear Vesting Contracts.
+
+## Test Suite Details
+
+### Tests Included
+
+-**Partial Unlock**: This property based test single-handedly confirms the proper initialization of Linear Vesting Contract and the successful unlocking of vested assets from the Contract at correct intervals. It does so by programmatically generating **100** test scenarios with entirely different Vesting datum values, unlock timings and their corresponding vested asset quantities. This confirms that the Contract allows accurate vesting of assets and their claim under varied conditions.
+
+## Running the Tests
+
+To execute the Linear Vesting Tests, you should follow the standard testing procedures outlined in the project documentation. Typically, this involves executing a command such as:
+
+```sh
+cabal new-test --test-show-details=streaming
+```
+
+This command will compile and execute all the test suites defined in the project. The output will show the status of each test case.
+
+### Test Outcome Summary
+
+In the most recent execution:
```

Comment:
We could add a link here to the project's github actions page: https://github.com/Anastasia-Labs/linear-vesting/actions